src/data/nat/choose.lean

```diff
@@ -100,8 +101,86 @@ theorem add_pow [comm_semiring α] (x y : α) (n : ℕ) :
 (commute.all x y).add_pow n
 
 /-- The sum of entries in a row of Pascal's triangle -/
-theorem sum_range_choose (n : ℕ) : 
+theorem sum_range_choose (n : ℕ) :
   ∑ m in range (n + 1), choose n m = 2 ^ n :=
 by simpa using (add_pow 1 1 n).symm
 
+/-
+Facts about more specific binomial coefficients and their sums
+-/
+
+-- This lemma exists only so that we can instantiate it with `i = m + 1`.
+private lemma reflect_sum_lemma (i : nat) (m : nat) (i_bound : i ≤ m + 1) (f : nat → nat)
+  (reflects : ∀ x ≤ 2 * m + 1, f x = f (2 * m + 1 - x)) :
+  ∑ j in (Ico (m + 1 - i) (m + 1)), f j = ∑ j in Ico (m + 1) (m + 1 + i), f j :=
+begin
+  induction i with i,
+  { simp },
+  {
+    have t : (m + 1) + (i + 1) = ((m + 1) + i) + 1, norm_num,
+    rw t, clear t,
+    rw <- (@sum_Ico_succ_top _ _ (m + 1) (m + 1 + i) (by exact nat.le.intro rfl) f).symm,
+    rw <- i_ih, clear i_ih,
+    have t : f (m + 1 + i) = f (m - i), by {
+      have munge : m + 1 + i ≤ 2 * m + 1,
```

Comment:
```suggestion
    have t : f (m + 1 + i) = f (m - i), by
    { have munge : m + 1 + i ≤ 2 * m + 1,
```

Reply:
I would hope that you can replace the proof of `munge` by a 1-line call to `omega`. But `omega` can be a nasty beast.

Reply:
Sadly it did not seem to be so; but I did replace much of it with `ring_exp`.